tests/unit/test_service.py

```diff
@@ -38,3 +44,117 @@ def test_install_apt_package_gpg_key_error(
 
     with pytest.raises(RuntimeError, match="Error installing the agent service"):
         charm._on_install(MagicMock(spec=ops.InstallEvent))
```

Comment:
Can you test by triggering this events instead?

Reply:
yes, I updated the code to trigger the event!